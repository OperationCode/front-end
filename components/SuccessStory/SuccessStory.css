.SuccessStory {
  display: flex;
  flex-flow: row nowrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  margin: 15px;
}

.image {
  margin: 15px;
  border-radius: 50%;
  border: 3px solid #41a4c0;
  width: 150px;
  max-width: 150px;
  height: 150px;
  max-height: 150px;
  flex: 1 0 auto;
}

.quote {
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-around;
  margin: 15px;
}

.quote > blockquote {
  font-size: 0.9rem;
  margin: 0;
  text-align: left;
}

.title {
  font-size: 1.7rem;
}

/* Vertical space between cards for vertical layout */
@media screen and (max-width: 650px) {
  .SuccessStory {
    flex-flow: row wrap;
  }
}
