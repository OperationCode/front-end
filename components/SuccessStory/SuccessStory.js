import React from 'react';
import { string } from 'prop-types';
import { Image } from '@innocuous/components';
import styles from './SuccessStory.module.css';

SuccessStory.propTypes = {
  imageSource: string.isRequired,
  quote: string.isRequired,
  title: string.isRequired,
};

function SuccessStory({ imageSource, quote, title }) {
  return (
    <div className={styles.SuccessStory}>
      <Image alt={`${title} headshot`} className={styles.image} src={imageSource} />
      <div className={styles.quote}>
        <h6 className={styles.title}>{title}</h6>
        <blockquote>{`"${quote}"`}</blockquote>
      </div>
    </div>
  );
}

export default SuccessStory;
