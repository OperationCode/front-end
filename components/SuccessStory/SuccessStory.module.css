.SuccessStory {
  box-sizing: border-box;
  display: flex;
  flex-flow: column nowrap;
  align-items: center;
  margin: 3rem 1rem;
  padding-top: 60px;
  position: relative;
  max-width: 320px;
  height: 670px;
}

.image {
  box-shadow: 1px 2px 4px 2px rgba(0, 0, 0, 0.2);
  position: absolute;
  top: 0;
  height: 200px;
  width: 200px;
}

.quote {
  display: flex;
  flex: 1;
  flex-flow: column nowrap;
  align-items: center;
  padding: 2rem;
  padding-top: 160px;
  border: 3px solid var(--primary);
}

.quote > blockquote {
  margin: 1rem 0;
  font-style: italic;
  font-size: 1rem;
}

.title {
  text-align: center;
  font-size: 1rem;
}

@media screen and (--medium-viewport) {
  .SuccessStory {
    height: 575px;
  }
}

@media screen and (--small-viewport) {
  .SuccessStory {
    height: auto;
  }
}
