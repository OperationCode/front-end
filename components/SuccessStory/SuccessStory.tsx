import { FlatCard } from 'components/Cards/FlatCard/FlatCard';

export interface SuccessStoryPropsType {
  /**
   * Path to image used on the card.
   */
  imageSource: string;
  /**
   * String applied tot he block quote element.
   */
  quote: string;
  /**
   * String applied to the card header.
   */
  title: string;
}

export function SuccessStory({ imageSource, quote, title }: SuccessStoryPropsType) {
  return (
    <FlatCard header={title} image={{ source: imageSource, alt: '' }}>
      <blockquote>{`“${quote}”`}</blockquote>
    </FlatCard>
  );
}
