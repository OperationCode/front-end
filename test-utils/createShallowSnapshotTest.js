import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers, components utilizing something that requires large
 * mocking, or very large components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
