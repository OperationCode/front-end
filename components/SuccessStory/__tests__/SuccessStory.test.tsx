import { SuccessStory } from '../SuccessStory';
import createShallowSnapshotTest from '@/test-utils/createShallowSnapshotTest';

describe('SuccessStory', () => {
  it('should render with required props', () => {
    createShallowSnapshotTest(
      <SuccessStory
        imageSource="image.png"
        quote="Best test of my life!"
        title="Great testing culture"
      />,
    );
  });
});
