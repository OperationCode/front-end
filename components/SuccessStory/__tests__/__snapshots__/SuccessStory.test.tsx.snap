// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SuccessStory > should render with required props 1`] = `
<FlatCard
  button={null}
  header="Great testing culture"
  image={
    {
      "alt": "",
      "source": "image.png",
    }
  }
>
  <blockquote>
    “Best test of my life!”
  </blockquote>
</FlatCard>
`;
